GetMessage/handler.ts

```diff
@@ -117,6 +127,47 @@ type IGetMessageHandler = (
 const LegalMessagePattern = t.interface({ legal_data: LegalData });
 type LegalMessagePattern = t.TypeOf<typeof LegalMessagePattern>;
 
+/**
+ * Checks whether the client service can read advanced message info (read_status and payment_Status)
+ */
+export const canReadAdvancedMessageInfo = (
+  message:
+    | ExternalCreatedMessageWithoutContent
+    | ExternalCreatedMessageWithoutContent,
+  authGroups: IAzureApiAuthorization["groups"]
+): boolean =>
+  message.feature_level_type === FeatureLevelTypeEnum.ADVANCED &&
+  authGroups.has(UserGroup.ApiMessageReadAdvanced);
+
+/**
+ * Checks whether the client service can read message read status
+ *
+ * @param serviceId the subscription id of the service
+ * @returns false if user revoked the permission to access the read status, true otherwise
+ */
+export const canReadMessageReadStatus = (_serviceId: ServiceId): boolean =>
+  false;
+
+/**
+ * Return a ReadStatusEnum
+ *
+ * @param maybeMessageStatus an Option of MessageStatus
+ * @returns READ if message status exists and isRead is set to true, UNREAD otherwise
+ */
+export const getReadStatusForService = (
+  maybeMessageStatus: O.Option<RetrievedMessageStatus>
+): ReadStatus =>
+  pipe(
+    maybeMessageStatus,
+    O.map(messageStatus =>
+      b.fold(
+        () => ReadStatusEnum.UNREAD,
+        () => ReadStatusEnum.READ
+      )(messageStatus.isRead)
+    ),
```

Comment:
```suggestion
    O.map(messageStatus => messageStatus.isRead),
    O.map(
      b.fold(
        () => ReadStatusEnum.UNREAD,
        () => ReadStatusEnum.READ
      )
    ),
```
maybe we could flatten the inner fold

Reply:
Way better, thanks. Done in [a0a83cf](https://github.com/pagopa/io-functions-services/pull/202/commits/a0a83cf33f3d0c9340dbc1283dd95052a4e48f3b)